src/pages/Authentication/components/Input/hooks/useTokenActions.ts

```diff
@@ -92,32 +68,47 @@ export const useTokenActions = () => {
         }
       }
     },
-    [setFieldError, chain, setFieldValue, setFieldTouched, setMetrics]
+    [chain, setFieldError, setMetrics]
   );
 
-  const handlePreventDefault = useCallback(
-    (event: FormEvent<HTMLDivElement>) => {
-      event.preventDefault();
-    },
-    []
+  const debouncedDecodeAndValidateToken = useMemo(
+    () => debounce(decodeAndValidateToken, 200),
+    [decodeAndValidateToken]
   );
 
-  const handleKeyDown = useCallback(
-    (e: React.KeyboardEvent<HTMLDivElement>) => {
-      const commonKey = e.ctrlKey || e.metaKey || e.shiftKey || e.altKey;
-      const allowedCombinationKeys = ['a', 'c', 'v', 'x'];
-      const arrowKeys = ['ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown'];
+  const handleChange = useCallback(
+    async (event: ChangeEvent<HTMLTextAreaElement> | string) => {
+      const token = typeof event === 'string' ? event : event.target.value;
 
-      const allowedCombinations =
-        commonKey && allowedCombinationKeys.includes(e.key);
-      const allowed =
-        commonKey || arrowKeys.includes(e.key) || allowedCombinations;
+      setFieldValue('token', token, false);
+      setFieldTouched('token', true, false);
 
-      if (!allowed) {
-        handlePreventDefault(e);
+      if (!token || token.match(/[\r\n]/gm) || token.match(/\s+/g)) {
```

Comment:
I find it hard to read token.match(/[\r\n]/gm), maybe isolate it into a constant?